### PR TITLE
Changed initial arm positions and rotations to be more accurate with VRChat

### PR DESCRIPTION
**Pull request by DSMT2**
After some comparisons with vrchat and cyan emu's tracking data positions, I have edited the initial starting positions and rotations for the arms on cyan emu to more closely fit with vrchat.

### CODE REVIEW

**Review by CyanLaser — COMMENTED** [review]
I did some checking on my own. It looks like CyanEmu's rotation is wrong and your value is correct, but the position appears to be avatar dependent. I could not get the same position offset you have with any avatar I tried. With 3 different avatars, the x value never changed for me and the y and z values varied slightly. Can you show what code you used to gather your data? CyanEmu's values should still be updated, but we should figure out what is the best average value to use. 

This is the code that I used. I have three transforms, with the left and right hands being a child of the head.
```
public override void PostLateUpdate()
{
    VRCPlayerApi player = Networking.LocalPlayer;
    var headTracking = player.GetTrackingData(VRCPlayerApi.TrackingDataType.Head);
    var leftHandTracking = player.GetTrackingData(VRCPlayerApi.TrackingDataType.LeftHand);
    var rightHandTracking = player.GetTrackingData(VRCPlayerApi.TrackingDataType.RightHand);
    
    headTransform.SetPositionAndRotation(headTracking.position, headTracking.rotation);
    leftHandTransform.SetPositionAndRotation(leftHandTracking.position, leftHandTracking.rotation);
    rightHandTransform.SetPositionAndRotation(rightHandTracking.position, rightHandTracking.rotation);

    text.text = "Head: " + headTransform.localPosition.ToString("F5") + ", " +
                headTransform.localRotation.eulerAngles.ToString("F0") + "\n" +
                "Left: " + leftHandTransform.localPosition.ToString("F5") + ", " +
                leftHandTransform.localRotation.eulerAngles.ToString("F0") + "\n" +
                "Right: " + rightHandTransform.localPosition.ToString("F5") + ", " +
                rightHandTransform.localRotation.eulerAngles.ToString("F0");
}
```

Screenshot 2021-07-27 041828
Screenshot 2021-07-27 042132
Screenshot 2021-07-27 042136

**Comment by DSMT2** [comment]
After some more checking on VRChat, I discovered that it is correct that the arm tracking positions are avatar dependent, as the numbers altered depending on what avatar was used.

I have used the testing code that you provided and added in an equation to calculate the avatar's height based on the difference between the head position and the player root position.

`    public override void PostLateUpdate()
    {
        VRCPlayerApi player = Networking.LocalPlayer;
        var headTracking = player.GetTrackingData(VRCPlayerApi.TrackingDataType.Head);
        var leftHandTracking = player.GetTrackingData(VRCPlayerApi.TrackingDataType.LeftHand);
        var rightHandTracking = player.GetTrackingData(VRCPlayerApi.TrackingDataType.RightHand);
        var playerPosition = player.GetPosition();
        var playerRotation = player.GetRotation();

        headTransform.SetPositionAndRotation(headTracking.position, headTracking.rotation);
        leftHandTransform.SetPositionAndRotation(leftHandTracking.position, leftHandTracking.rotation);
        rightHandTransform.SetPositionAndRotation(rightHandTracking.position, rightHandTracking.rotation);
        playerPosTransform.SetPositionAndRotation(playerPosition, playerRotation);

        var headPos = headTransform.localPosition;
        var avatarHeight = headPos - playerPosTransform.localPosition;
        var rightHandDifference = rightHandTransform.localPosition - avatarHeight;
        var leftHandDifference = leftHandTransform.localPosition - avatarHeight;

        text.text = "Head = " + headPos.ToString("F5") + "\nHeight = " + avatarHeight.y.ToString("F5") + "\nRight = " + rightHandDifference.ToString("F5") + "\nLeft = " + leftHandDifference.ToString("F5");

        //text.text = "Head: " + headTransform.localPosition.ToString("F5") + ", " +
        //            headTransform.localRotation.eulerAngles.ToString("F0") + "\n" +
        //            "Left: " + leftHandTransform.localPosition.ToString("F5") + ", " +
        //            leftHandTransform.localRotation.eulerAngles.ToString("F0") + "\n" +
        //            "Right: " + rightHandTransform.localPosition.ToString("F5") + ", " +
        //            rightHandTransform.localRotation.eulerAngles.ToString("F0");
    }`

TestingNumbers

File: CyanEmu/Scripts/CyanEmuPickupHelper.cs [hunk 1]
```diff
@@ -10,7 +10,7 @@ public class CyanEmuPickupHelper : MonoBehaviour, ICyanEmuInteractable
         private const float INITIAL_PICKUP_DURATION_ = 0.5f;
         private const float MAX_PICKUP_DISTANCE_ = 0.25f;
         private static Quaternion GRIP_OFFSET_ROTATION_ = Quaternion.Euler(0, 45, 0);
```
Comment (CyanLaser) [inline CyanEmu/Scripts/CyanEmuPickupHelper.cs:12]: I just checked the rotations in VRChat. Your update to Gun offset is correct, but it looks like Grip also needs to be modified. VRChat reports 35 degrees relative to the hand tracker. Other than that, everything else is good to be merged in. 

```
Vector3 gripRelPos = Vector3.zero;
Quaternion gripRelRot = Quaternion.identity;
if (gripPickup.IsHeld)
{
    Transform gripPickupTransform = gripPickup.transform;
    Transform prevGripPar = gripPickupTransform.parent;
    gripPickupTransform.SetParent(rightHandTransform, true);
    gripRelPos = gripPickupTransform.localPosition;
    gripRelRot = gripPickupTransform.localRotation;
    gripPickupTransform.SetParent(prevGripPar, true);
}
```
Screenshot 2021-07-28 154932

**Comment by DSMT2** [comment]
I have modified the grip rotation to 35 on the y axis, based on my test it should match VRChat's grip rotation for pickups.

**Pull request merged**